ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp

```diff
@@ -0,0 +1,2613 @@
+                                                        .pZCoord[nIVrtAcum]);
+                                }
+                                else
+                                {
+                                    poRing.addPoint(phMiraMonLayer->ReadFeature
+                                                        .pCoord[nIVrtAcum]
+                                                        .dfX,
+                                                    phMiraMonLayer->ReadFeature
+                                                        .pCoord[nIVrtAcum]
+                                                        .dfY);
+                                }
+
+                                nIVrtAcum++;
+                            }
+                            poP->addRing(&poRing);
+                        }
+                    }
+                }
+
+                break;
+        }
+
+        if (poGeom == nullptr)
+            return nullptr;
+    }
+
+    /* -------------------------------------------------------------------- */
+    /*      Create feature.                                                 */
+    /* -------------------------------------------------------------------- */
+    OGRFeature *poFeature = new OGRFeature(poFeatureDefn);
+    if (poGeom)
+    {
+        poGeom->assignSpatialReference(m_poSRS);
+        poFeature->SetGeometryDirectly(poGeom);
+    }
+
+    /* -------------------------------------------------------------------- */
+    /*      Process field values if its possible.                           */
+    /* -------------------------------------------------------------------- */
+    if (phMiraMonLayer->pMMBDXP &&
+        (MM_EXT_DBF_N_RECORDS)nFeatureId < phMiraMonLayer->pMMBDXP->nRecords)
+    {
+        MM_EXT_DBF_N_FIELDS nIField;
+
+        for (nIField = 0; nIField < phMiraMonLayer->pMMBDXP->nFields; nIField++)
+        {
+            MMResizeStringToOperateIfNeeded(
+                phMiraMonLayer,
+                phMiraMonLayer->pMMBDXP->pField[nIField].BytesPerField);
+
+            if (poFeature->GetDefnRef()->GetFieldDefn(nIField)->GetType() ==
+                    OFTStringList ||
+                (poFeature->GetDefnRef()->GetFieldDefn(nIField)->GetType() ==
+                     OFTString &&
+                 poFeature->GetDefnRef()->GetFieldDefn(nIField)->GetSubType() ==
+                     OFSTJSON))
+            {
+                if (poFeature->GetDefnRef()
+                        ->GetFieldDefn(nIField)
+                        ->GetSubType() == OFSTJSON)
+                {
+                    // REVISAR
+                    MMResizeStringToOperateIfNeeded(
+                        phMiraMonLayer,
+                        phMiraMonLayer->pMMBDXP->BytesPerRecord +
+                            2 * phMiraMonLayer->pMultRecordIndex[nIElem].nMR +
+                            8);
+                    strcpy(phMiraMonLayer->szStringToOperate, "(``[");
```

Comment:
I'm confused. "(``[" is not a valid format for a JSON array. This should be just [a,b,c,d]. Cf https://www.w3schools.com/js/js_json_arrays.asp .
and instead of using C string operations, you could just use std::string and enjoy safe concatenation without risk of overflows of the buffer

Reply:
Done. You can review but the reading from a file part is the same, but the part of the construction of the final string is easier to understand, I hope.
If you find something that can be improved go ahead.
I've also converted ``[ to a simple [ (I suppose I read it on a bad place )